core/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/BeanValidationScanner.java

```diff
@@ -340,8 +340,9 @@ void negative(AnnotationTarget target, Schema schema) {
         if (constraint != null && schema.getMaximum() == null) {
             Boolean exclusive = schema.getExclusiveMaximum();
 
-            if (exclusive != null && exclusive) {
+            if (exclusive == null || exclusive) {
                 schema.setMaximum(BigDecimal.ZERO);
+                schema.setExclusiveMaximum(Boolean.TRUE);
```

Comment:
I think a null/undefined value should be interpreted as false.

https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-5.3

Reply:
Disregard - I did not consider non-integer cases for positive/negative, so your change makes sense to me now.